File: homl-web/src/internal/domain/person.go
package domain

type Person struct {
	Id         uint   `json:"id"`
	Firstname  string `json:"firstname"`
	Lastname   string `json:"lastname"`
	IdCategory string `json:"idCategory"`
}

type Nickname struct {
	Id       uint   `json:"id"`
	Nickname string `json:"nickname"`
}

type GetPersonsResponse struct {
	Person
	Nicknames []Nickname `json:"nicknames"`
}

File: homl-ui/lib/data/models/person.dart
import 'package:json_annotation/json_annotation.dart';
import 'nickname.dart';

part 'person.g.dart';

@JsonSerializable(explicitToJson: true)
class Person {
  final String firstname, lastname;
  final List<Nickname> nicknames;

  Person(
      {required this.firstname,
      required this.lastname,
      required this.nicknames});

  factory Person.fromJson(Map<String, dynamic> json) => _$PersonFromJson(json);

  Map<String, dynamic> toJson() => _$PersonToJson(this);
}
